lib/ecmascript.ts

```diff
@@ -4830,16 +4793,10 @@ export function RoundDuration(
       const weeksInDays = DaysUntil(yearsMonthsLater, yearsMonthsWeeksLater);
       // This conversion is needed because of an issue in the spec that's likely to change.
       // See https://github.com/tc39/proposal-temporal/issues/1821.
-      relativeTo = CreateTemporalDateTime(
+      relativeTo = CreateTemporalDate(
```

Comment:
I found the cause of some of the test failures: in place of this line, proposal-temporal has this code:
```
      relativeTo = yearsMonthsLater;
```

Iin https://github.com/js-temporal/temporal-polyfill/commit/2c1fbb58d471a8ee041a1e6dee92285981849139, `relativeTo` is never reassigned here, but (at least according to proposal-temporal) it should be reassigned here.

There's still two failing tests related to `'year'` units. I'll take a look at those next.

Reply:
Thanks Justin! This really helped - I diffed the content of the RoundDuration method for both implementations and the reassignment here turned out to be critical - I needed to change all usages of `relativeToDateOnly` back to `relativeTo` along with this reassignment and then all the Demitasse tests pass.

Really appreciate you diving in to take a look at this for me.

Reply:
Yep. Originally, `relativeToDateOnly` was there because TS emitted a compiler error when passing `relativeTo` to some other method (forget which) that required a PlainDate.  Looks like that error doesn't show up anymore, perhaps because that method was changed to also accept a PlainDateLike too?  Regardless, glad this helped.